include/mapbox/geojson_value_impl.hpp

```diff
@@ -0,0 +1,295 @@
+    }
+
+    auto coordinatesIt = valueObject->find("coordinates");
+    if (coordinatesIt == valueObject->end()) {
+        throw error(typeString + " geometry must have a coordinates property");
+    }
+
+    const auto *coordinateArray = coordinatesIt->second.getArray();
+    if (!coordinateArray) {
+        throw error("coordinates property must be an array");
+    }
+
+    if (typeString == "Point")
+        return geometry{ convert<point>(*coordinateArray) };
+    if (typeString == "MultiPoint")
+        return geometry{ convert<multi_point>(*coordinateArray) };
+    if (typeString == "LineString")
+        return geometry{ convert<line_string>(*coordinateArray) };
+    if (typeString == "MultiLineString")
+        return geometry{ convert<multi_line_string>(*coordinateArray) };
+    if (typeString == "Polygon")
+        return geometry{ convert<polygon>(*coordinateArray) };
+    if (typeString == "MultiPolygon")
+        return geometry{ convert<multi_polygon>(*coordinateArray) };
+
+    throw error(typeString + " not yet implemented");
+}
+
+template <>
+feature convert<feature>(const value &val) {
+    auto *valueObject = val.getObject();
+    if (!valueObject) {
+        throw error("GeoJSON must be an object");
+    }
+
+    auto typeIt = valueObject->find("type");
+    if (typeIt == valueObject->end()) {
+        throw error("Feature must have a type property");
+    }
+
+    const auto &typeValue = typeIt->second;
+    if (!typeValue.is<std::string>()) {
+        throw error("Feature 'type' property must be of a String type");
+    }
+
+    if (*typeValue.getString() != "Feature") {
+        throw error("Feature type must be Feature");
+    }
+
+    auto geometryIt = valueObject->find("geometry");
+    if (geometryIt == valueObject->end()) {
+        throw error("Feature must have a geometry property");
+    }
+
+    feature result{ convert<geometry>(geometryIt->second) };
+    auto idIt = valueObject->find("id");
+    if (idIt != valueObject->end()) {
+        result.id =
+            idIt->second.match([](const std::string &string) -> identifier { return { string }; },
+                               [](int64_t number) -> identifier { return { number }; },
+                               [](uint64_t number) -> identifier { return { number }; },
+                               [](double number) -> identifier { return { number }; },
+                               [](const auto &) -> identifier {
+                                   throw error("Feature id must be a string or number");
+                               });
+    }
+
+    auto propertiesIt = valueObject->find("properties");
+    if (propertiesIt != valueObject->end() &&
+        !propertiesIt->second.is<mapbox::geojson::null_value_t>()) {
+        if (!propertiesIt->second.is<value::object_type>()) {
+            throw error("properties must be an object");
+        }
+        result.properties = *propertiesIt->second.getObject();
+    }
```

Comment:
I think in the GeoJSON spec the feature must have a `properties` property.

```
   A Feature object represents a spatially bounded thing.  Every Feature
   object is a GeoJSON object no matter where it occurs in a GeoJSON
   text.

   o  A Feature object has a "type" member with the value "Feature".

   o  A Feature object has a member with the name "geometry".  The value
      of the geometry member SHALL be either a Geometry object as
      defined above or, in the case that the Feature is unlocated, a
      JSON null value.

   o  A Feature object has a member with the name "properties".  The
      value of the properties member is an object (any JSON object or a
      JSON null value).
```

Reply:
Thanks, updating PR.

Reply:
Actually, this is value->feature conversion, feature have properties by default, no change needed.